src/components/ConnectWallet/ConnectWalletDialog.tsx

```diff
@@ -46,18 +58,93 @@ interface ButtonProps {
   walletName?: string
   logoSrc?: string
   caption?: string
+  connection?: Web3Connection
   onClick: () => void
 }
 
-function WalletConnectButton({ walletName, logoSrc, caption, onClick }: ButtonProps) {
+const wcQRUriAtom = atom<string | undefined>(undefined)
+
+function WalletConnectButton({ walletName, logoSrc, caption, connection: wcTileConnection, onClick }: ButtonProps) {
+  const [tileConnector] = wcTileConnection as [WalletConnect, Web3ReactHooks]
+  const [error, setError] = useState(undefined)
+
+  const [QRUri, setQRUri] = useAtom(wcQRUriAtom)
+  const [qrCodeSvg, setQrCodeSvg] = useState<string>('')
+
+  useEffect(() => {
+    if (QRUri) {
+      formatQrCodeImage(QRUri)
+    } else {
+      tileConnector.activate().catch(setError)
+    }
+  }, [QRUri, tileConnector])
+
+  useEffect(() => {
+    // Log web3 errors
+    if (error) {
+      console.error('web3 error:', error)
+    }
+  }, [error])
+
+  useEffect(() => {
+    const disconnectListener = async (err: Error | null, _: any) => {
+      if (err) console.warn(err)
+      // Clear saved QR URI after disconnection
+      setQRUri(undefined)
+      tileConnector.deactivate()
+    }
+    tileConnector.provider?.connector.on('disconnect', disconnectListener)
+
+    // Need both URI event listeners
+    tileConnector.events.on(URI_AVAILABLE, async (uri: string) => {
+      if (uri) {
+        setQRUri(uri)
+        await formatQrCodeImage(uri)
+      }
+    })
+
+    const uriListener = async (err: Error | null, payload: any) => {
+      if (err) console.warn(err)
+      const uri: string = payload.params[0]
+      if (uri) {
+        setQRUri(uri)
+        await formatQrCodeImage(uri)
+      }
+    }
+    tileConnector.provider?.connector.on('display_uri', uriListener)
+
+    return () => {
+      tileConnector.events.off(URI_AVAILABLE)
+      ;(tileConnector.provider?.connector as unknown as EventEmitter | undefined)?.off('display_uri', uriListener)
+    }
+  })
+
+  async function formatQrCodeImage(uri: string) {
+    let result = ''
+    const dataString = await QRCode.toString(uri, { margin: 0, type: 'svg' })
```

Comment:
The QRCode that we end up with is not as pretty as the WC QR code. Do you know why?

Reply:
I think what you have currently complies with the designs: https://www.figma.com/file/A7w2snzxyFK2qj2aFpkkm1/Widgets?node-id=6057%3A39728

Reply:
The colors on the QR code should match the theme, and not just be white/black.
I'm fine with this being done in a follow-up PR, but it should be done before merging to `main`.